packages/expo-location/android/src/main/java/expo/modules/location/LocationModule.kt

```diff
@@ -501,27 +508,54 @@ class LocationModule : Module(), LifecycleEventListener, SensorEventListener, Ac
   }
 
   private fun startHeadingUpdate() {
-    val locationControl = SmartLocation.with(mContext).location().oneFix().config(LocationParams.BEST_EFFORT)
-    val currLoc = locationControl.lastLocation
-    if (currLoc != null) {
+    val locationManager = mContext.getSystemService(Context.LOCATION_SERVICE) as LocationManager
+    if (ActivityCompat.checkSelfPermission(mContext, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED &&
+      ActivityCompat.checkSelfPermission(mContext, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED
+    ) {
+      return
+    }
+    val lastLocation =
+      locationManager.getLastKnownLocation(LocationManager.GPS_PROVIDER)
+        ?: locationManager.getLastKnownLocation(LocationManager.NETWORK_PROVIDER)
+
+    if (lastLocation != null) {
       mGeofield = GeomagneticField(
-        currLoc.latitude.toFloat(), currLoc.longitude.toFloat(), currLoc.altitude.toFloat(),
+        lastLocation.latitude.toFloat(),
+        lastLocation.longitude.toFloat(),
+        lastLocation.altitude.toFloat(),
         System.currentTimeMillis()
       )
     } else {
-      locationControl.start { location: Location ->
-        mGeofield = GeomagneticField(
-          location.latitude.toFloat(), location.longitude.toFloat(), location.altitude.toFloat(),
-          System.currentTimeMillis()
-        )
+      val locationRequest = LocationRequest.Builder(
```

Comment:
smart-location seems support fallback when fused location provider (aka google play services) is not available on devices. then it fallbacks to AOSP LocaitonManager: https://github.com/mrmans0n/smart-location-lib?tab=readme-ov-file#changing-providers

is it fine for us to support only when google play services available?

Reply:
is default the LocationGooglePlayServicesWithFallbackProvider?

Reply:
Yes, we didn't provide one so it would have used that.